microservices/embeddingservice/dto/embed_data.py

```diff
@@ -1,6 +1,7 @@
 from pydantic import BaseModel
+from typing import Optional
 
 
 class EmbedData(BaseModel):
-    id: str
+    id: Optional[str] = ""
```

Comment:
FYI Optional means nullable
https://docs.python.org/3/library/typing.html#typing.Optional

since you provide a default the actually is of type string as None will always be overwritten by the default